### DISCUSSION TITLE
Reading and theme settings should be per device

### DISCUSSION THREAD

**Discussion by chnorton** [Kareadita/Kavita#2929] [Ideas]
### Idea Description

I've recently started using an e-ink tablet, which is much nicer to view using the Light theme. On all other devices (desktop, phone) the Dark theme is better. Unfortunately the theme setting is locked to the account, rather than the account + device, so is applied across all devices and doesn't (as far as I can tell) have an option of only setting a theme on a specific device.

Similarly, the reading settings would also be better to be per device, since what looks comfortable might vary based on the screen you're looking at it on. (This is honestly less of an issue than the theme.)

I think ideally you'd have a descending hierarchy of how these settings are applied:

- Global defaults
- Account setting
- Device override

The UI probably doesn't need to change - the existing "Use Default" could set the default for the account, and "Apply" only applies the setting for the device. Storing the settings in localStorage or similar would probably work for keeping them on a specific device; since they're per device I don't think they'd have to be saved back to the server.

Hopefully I haven't missed something and it's already possible to do this in some way!

### Idea Category

User Experience

### Duration of Using Kavita

1.5 years

### Before submitting

- [X] I've already searched for existing ideas before posting.

**Comment by rix8675** [comment]
this would be really useful. i swap between my pc/tablet/phone quite often.

**Comment by Neshura87** [comment]
Haven't taken a look at the codebase but couldn't this be implemented quite easily by simply storing and loading the theme to/from a cookie or local storage instead of always fetching it from the server?

**Reply by Neshura87** [reply]
I'll just dump my thoughts about this here, probably will open a PR along the lines of it later:
- always fetch default theme from the server (as is currently)
- if localstorage does not contain a theme *or* the localstorage theme is "default" apply the account default theme (this would also be as is currently)
- if localstorage *does* contain a theme *and* the theme is not "default" then ignore the account default and use the set theme
-- if the localstorage theme does not exist fall back to the account default
- when applying a theme if the theme applied is equal to the account default then instead of the theme name store "default" in localstorage -> on every device where the default theme is used switching the theme on any device will also change it on the others, this behaves as is currently the case. However now we can also set themes on a per device basis without changing the default every time.

**Reply by Neshura87** [reply]
Note: looks like the "Set Default" button might set the theme default site wide, would need another button or a drop down option for admins if that is the case

**Comment by YodaDaCoda** [comment]
A pattern I've noticed is the ability to configure separate dark and light mode themes, which are automatically toggled based on the device's theme preference. I think this provides a cleaner user experience compared to an account-level setting with device overrides.

**Comment by uwuceo** [comment]
This would be very helpful. When I read webtoons and I switch between reading on my phone and my PC I have to change the width override on each device between off and 35% to see about the same image height.

**Reply by Fesaa** [reply]
Heya, you can use the disable width override setting for this.

<img width="896" height="235" alt="image" src="https://github.com/user-attachments/assets/5b6de276-de02-47a6-9820-af0c6e061cb0" />

**Reply by uwuceo** [reply]
That is the perfect setting, I don't know how I didn't notice it, thank you!

**Comment by ark4d-a** [comment]
> I would add to that Similarly, the reading settings would also be better to be per device, since what looks comfortable might vary based on the screen you're looking at it on. **(This is honestly less of an issue than the theme.)**

I would no agree. I read epub books and it's painful to switch between PC and phone. What looks good on PC doesnt on the phone. See examples: 

On PC (looks okay): 
<img width="400" height="400" alt="image" src="https://github.com/user-attachments/assets/c6544d32-f535-44c9-b27f-980221b5ff32" />

On phone (too much empty spaces right and left): 
<img width="385" height="640" alt="image" src="https://github.com/user-attachments/assets/c2bd6408-facf-41fb-9af5-3043b1a8d624" />


It would be nice if my epub settings would be just stored locally. I dont think its a problem to set them up once per device.

**Comment by majora2007** [comment]
This is delivered as of v0.8.8.19 and will be out in v0.8.9 (stable).
